docs/reproducing-results.md

```diff
@@ -362,9 +362,88 @@ run inference through Nvidia NIM API.
        subprocess.run(cmd, shell=True, check=True)
    ```
 
-6. Now all the data is generated and you can follow up by converting it to the SFT format.
-
-   TBD
+6. Check for test set contamination.
+   We test against GSM8K, MATH, AMC 2023, and AIME 2024.  
+ 
+   Retrieve top-5 similar items from the test sets
+   ```
+   python -m nemo_skills.inference.retrieve_similar \
+      ++retrieve_from="./nemo_skills/dataset/gsm8k/test.jsonl ./nemo_skills/dataset/math/test.jsonl ./nemo_skills/dataset/amc23/test.jsonl ./nemo_skills/dataset/aime24/test.jsonl" \
+      ++compare_to="<path to workspace>/new-problems-solution-augmentation/**/output-rs0.jsonl" \
+      ++output_file=<path to workspace>/new-problems-solution-augmentation/contamination-retrieved.jsonl \
+      ++top_k=5
+   ```
+   > **_NOTE:_** Currently the above command doesn't run inside docker, so you will need to install additional packages.
+
+   Next, you need to run LLM inference to check those closest found problems from the output file. We use the Llama3.1-405B-Instruct model for this, and here's one way of doing it via Nvidia API catalog.
+
+    ```
+    ns check_contamination \
+        --cluster=local \
+        --input_file=<path to workspace>/new-problems-solution-augmentation/contamination-retrieved.jsonl \
```

Comment:
```suggestion
        --input_file=/workspace/new-problems-solution-augmentation/contamination-retrieved.jsonl \
```
the reason we need <path to workspace> is that because retrieve-similar doesn't yet work as a normal pipeline script and so we need "unmounted" path there. But all other commands should still use /workspace